lib/nostrum/shard/dispatch.ex

```diff
@@ -140,7 +149,12 @@ defmodule Nostrum.Shard.Dispatch do
 
     # Starting this as a task to avoid blocking the dispatch process
     # since this is a potentially long operation
-    Task.start(fn -> MessageCache.channel_delete(channel_id) end)
+    func = fn ->
+      MessageCache.channel_delete(channel_id)
+    end
+
+    {:ok, _pid} =
+      Bot.spawn_task(func)
```

Comment:
So the spawned task doesn't need to fall back on ancestor tracking
```suggestion
    bot_name = state.bot_options.name
    
    func = fn ->
      Bot.set_bot_name(bot_name)
      MessageCache.channel_delete(channel_id)
    end

    {:ok, _pid} = Bot.spawn_task(func, bot_name, state.seq)
```